plugins/database/src/db_plugin.cpp

```diff
@@ -15599,6 +15599,60 @@ auto db_data_object_finalize_op(irods::plugin_context& _ctx, const char* _json_i
     return SUCCESS();
 } // db_data_object_finalize_op
 
+auto db_check_auth_credentials_op(irods::plugin_context& _ctx,
+                                  const char* _username,
+                                  const char* _zone,
+                                  const char* _password,
+                                  int* _would_succeed) -> irods::error
+{
+    if (const auto ret = _ctx.valid(); !ret.ok()) {
+        return PASS(ret);
+    }
+
+    if (!_username || !_zone || !_password || !_would_succeed) {
+        log_db::error("{}: Received one or more null pointers.", __func__);
+        return ERROR(SYS_INVALID_INPUT_PARAM, "Received one or more null pointers.");
+    }
+
+    *_would_succeed = -1; // Indicates the correctness of the creds is unknown.
+
+    std::array<char, MAX_PASSWORD_LEN + 20> decoded_password{};
```

Comment:
+20 for that hard-coded string?

Reply:
I wish the reason for the `+ N` was obvious so we can give it a name, but at this point, IDK if it's even worth it.

I'm on the side of perhaps we just hide all of this so no one ever has to see it.

Thoughts?

_Am I misunderstanding?_

Reply:
Yep, I think I misunderstood the question.

You're asking if the +20 is for the hard-coded string known by the server and the clients, correct?

Reply:
Oh, yes, sorry. That was a real question because I wanted to make sure I understood what it meant. A comment or something would of course be helpful, but that was mainly for my own understanding.

Reply:
I really don't know what the +20 is for.

If it helps, I pretty much copied what I saw the existing functions doing when handling passwords.

For now, we can create an issue to investigate the hard-coded integers and handle it later.

Reply:
Created issue for this. See #7224.